vico/core/src/main/java/com/patrykandpatrick/vico/core/chart/draw/ChartDrawContextExtensions.kt

```diff
@@ -123,6 +125,13 @@ public fun <Model : ChartEntryModel> ChartDrawContext.drawMarker(
                 )
                 setWasMarkerVisible(true)
             }
+            if (lastMarkerEntryModels.containsAll(markerEntryModels).not()) {
```

Comment:
I’d suggest requiring `wasMarkerVisible` to be true here, so only one function is called when a marker appears. Also, while `containsAll` is most likely sufficient, I’d say it’d be preferable to check if the content of the two lists is identical.

```suggestion
            if (wasMarkerVisible && markerEntryModels != lastMarkerEntryModels) {
```